parachain/pallets/outbound-queue/src/lib.rs

```diff
@@ -314,94 +397,46 @@ pub mod pallet {
 			T::GasMeter::MAXIMUM_BASE_GAS + T::GasMeter::maximum_required(command)
 		}
 
-		/// Fee in DOT for delivering a message.
-		pub(crate) fn delivery_fee(command: &Command) -> T::Balance {
-			let max_gas_used = Self::maximum_overall_required_gas(command);
-			T::DeliveryFeePerGas::get().saturating_mul(max_gas_used.into())
-		}
-
-		/// Maximum refund in Ether for delivering a message
-		pub(crate) fn maximum_refund(command: &Command) -> u128 {
-			let max_gas_used = Self::maximum_overall_required_gas(command);
-			T::DeliveryRefundPerGas::get().saturating_mul(max_gas_used as u128)
-		}
-	}
-
-	impl<T: Config> OutboundQueueTrait for Pallet<T> {
-		type Ticket = OutboundQueueTicket<MaxEnqueuedMessageSizeOf<T>>;
-		type Balance = T::Balance;
-
-		fn validate(message: &Message) -> Result<(Self::Ticket, Fees<Self::Balance>), SendError> {
-			// The inner payload should not be too large
-			let payload = message.command.abi_encode();
-
-			// Create a message id for tracking progress in submission pipeline
-			let message_id: MessageHash = sp_io::hashing::blake2_256(&(message.encode())).into();
-
-			ensure!(
-				payload.len() < T::MaxMessagePayloadSize::get() as usize,
-				SendError::MessageTooLarge
+		/// Calculate fee in native currency for delivering a message.
+		pub(crate) fn calculate_fee(command: &Command) -> Option<Fee<T::Balance>> {
+			let max_gas = Self::maximum_overall_required_gas(command);
+			let remote_fee = Self::calculate_remote_fee(
+				max_gas,
+				Self::fee_config().fee_per_gas,
+				Self::fee_config().reward,
```

Comment:
Since we need to config `fee_per_gas` and `reward` anyway, can we make them `DOT` based then no need to introduce the `swap_ratio` config?

Reply:
If we add `exchange_rate` into the calculation, we also need to take token precision into consideration, DOT is 10 while Ether 18.

Reply:
Notice that reward in Ether is required anyway pay for the relayer so `exchange_rate`  seems inevitable?

Reply:
Yes, exactly. I missed that in the previous PRs.